contracts/src/selector.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2025 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use hex::FromHex;
+use risc0_zkvm::sha::Digest;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+#[non_exhaustive]
+pub enum SelectorError {
+    #[error("Unsupported selector")]
+    UnsupportedSelector,
+    #[error("Selector (0) does not have verifier parameters")]
```

Comment:
I think this needs to be curly braces. I think.
```suggestion
    #[error("Selector {0} does not have verifier parameters")]
```